src/compiler/commandLineParser.ts

```diff
@@ -545,6 +546,32 @@ namespace ts {
             }
             return fileNames;
         }
+
+        function getTypingOptions(): TypingOptions {
+            const options: TypingOptions =
+                getBaseFileName(configFileName) === "jsconfig.json" ? { enableAutoDiscovery: true } : undefined;
+            const jsonTypingOptions = json["typingOptions"];
+            if (jsonTypingOptions) {
+                for (const id in jsonTypingOptions) {
+                    if (id === "enableAutoDiscovery") {
+                        if (typeof jsonTypingOptions[id] === "boolean") {
+                            options.enableAutoDiscovery = jsonTypingOptions[id];
+                        }
+                    }
+                    else if (id === "include") {
+                        options.include = jsonTypingOptions[id] instanceof Array ? <string[]>jsonTypingOptions[id] : [];
+                    }
+                    else if (id === "exclude") {
+                        options.exclude = jsonTypingOptions[id] instanceof Array ? <string[]>jsonTypingOptions[id] : [];
+                    }
+                    else {
+                        // Todo: change diagnostics

```

Comment:
I don't understand what this TODO means? If you are not going to do it this code review, please make any it clearer for anyone else.